### PR TITLE
types/time: optimize time convertDateFormat function by remove fprintf

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Optimize time convertDateFormat function by remove fprintf.

* BenchMark `DateFormat` function
```shell
// before
BenchmarkFormat-12       2000000               637 ns/op
// This PR
BenchmarkFormat-12      10000000               230 ns/op
```

```sql 
create table t (a timestamp);
-- insert 100W rows data;
-- bench select.
select * from t;

// before
select * from t;    1.2S


// This PR
select * from t;      0.9S
```

### Check List 

Tests 

 - No Test. Tiny refactor.

Code changes

 - Has exported function/method change

Side effects

Related changes

### CODE REVIEW

**Comment by crazycs520** [comment]
@tiancaiamao @bb7133 PTAL

File: types/time.go [hunk 1]
```diff
@@ -2089,6 +2090,26 @@ func (t Time) convertDateFormat(b rune, buf *bytes.Buffer) error {
...
 }
 
+func formatIntWidth2(i int) string {
+	if i < 10 {
+		return "0" + strconv.FormatInt(int64(i), 10)
+	}
+	return strconv.FormatInt(int64(i), 10)
+}
+
+func formatIntWidth4(i int) string {
```
Comment (winkyao) [inline types/time.go]: Could you implement a `formatIntWidth` function, and pass the width as a parameter?

File: types/time.go [hunk 2]
```diff
@@ -2089,6 +2090,19 @@ func (t Time) convertDateFormat(b rune, buf *bytes.Buffer) error {
 	return nil
 }
 
+// FormatIntWidthN uses to format int with width.
```
Comment (lzmhhh123) [inline types/time.go]:
```suggestion
// FormatIntWidthN uses to format int with width. Insufficient digits are filled by 0.
```

**Pull request merged**